### DISCUSSION TITLE
Welcome to papyrus Discussions!

### DISCUSSION THREAD

**Discussion by dan-starkware** [starkware-libs/papyrus#380] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by ChiHaoLu** [comment]
Hi, I have some questions about the full node in StarkNet.  We can observe that a Full Node can synchronize its state from three sources: the Sequencer, P2P (other Full Nodes), and Ethereum. If the contents of these three sources differ, how would a Full Node decide the current state?

Based on my understanding:

1. When a Full Node synchronizes a new transaction pending from the Sequencer, it adds this new transaction to the Pending state.
1. Once the Sequencer successfully executes the transaction and obtains the state_diff, the transaction enters the Accepted on Layer2 state. At this point, the new state will be updated in the Local DB of the running Node.
1. After the Layer1 Core (Rollup) Contract updates the state_diff, the transaction enters the Accepted on Layer1 state.

To delve deeper into the three sync data methods, they represent different implications:

- Requesting data from the Sequencer Gateway is the first step to obtain the current transaction data, which means acquiring the "most recent state (block)" on Layer2.
- Requesting data from Ethereum and performing verification is done to ensure the accuracy of the final data, which means obtaining the "latest confirmed Layer2 state (block)" on Layer1.
- Conducting P2P sync with other Layer2 Full Nodes is aimed at expediting the synchronization process of historical data and ensuring network consistency.

I believe there might be inaccuracies in my description, and I'm unsure about the correct approach. I would appreciate it if you could correct my description. Thank you very much!

**Reply by iliav-starkware** [reply]
Howdy @ChiHaoLu!

- A node's local view of L2 may (in unhappy flows) conflict with its local view of L1. For Starknet, information from L1 always takes precedence. Here's an example. First, the Starknet L1 rollup contract has a block hash and a block number (stored as "sequence number"). Suppose the block number on the L1 rollup contract is 2550. Suppose you received L2-block #3000 and then synched backwards to L2-block #2500, only to discover that its block hash conflicts with the L1 rollup contract. In this case, you would dismiss the L2-blocks you synched on, and try to sync forward from the L1 state.
- There is no P2P network at the time of writing.
- In the future, the P2P network will not be used only for synching on historical data, but rather for listening to the network in real time. Specifically, the sequencer gateway will be deprecated, with information being broadcast by the sequencer to some peers, and then propagated throughout the network e.g by gossip.

Does this answer your questions?

**Comment by jacketchen91** [comment]
![微信截图_20230914223454](https://github.com/starkware-libs/papyrus/assets/92106237/10b54d10-57c8-471d-8c07-d38056ccf808)
Hello, administrator, when I tried to use the command to install the borrow point, an error occurred, saying that the image file could not be found. What is wrong? I installed papyrus according to the tutorial on github

**Comment by kompotkot** [comment]
Hello, from README.md not clear where to set `node_url`. As I understand it should be `.toml` file because I am getting an error:
```
Caused by:
  TOML parse error at line 1, column 10
  ...
  invalid string
  expected `"`, `'`
```
when providing it as argument `cargo run --release --package papyrus_node --bin papyrus_node --config node_url="https://123"`.
But all config presets in `json` format.

**Reply by kompotkot** [reply]
Also there are not much `.toml` files in directory when I search with `find . -type f -name '*.toml'`.

**Comment by Th0rgal** [comment]
Hey, I can't find a way to open a discussion on this repository

**Comment by TropicolX** [comment]
Hi @dan-starkware,
My name is Oluwabusayo. I'm a Technical writer.
I love diving into open-source projects. I recently came across Papyrus and was really impressed by its offering as a node client for Starknet. However, I noticed it doesn't have a documentation site yet.

I’d love to help out by creating a comprehensive documentation site for the tool. Right now, there’s just the GitHub README, and I think we can do a lot more to make Papyrus more user-friendly and accessible.

Here’s what I’m thinking:
- Detailed Documentation: Extract and expand the current Github README into a full documentation site using [Docusaurus](https://docusaurus.io/) with clear sections on installation, setup, and usage. This would be similar to [Starknet's official documentation](https://docs.starknet.io/documentation/) or competitors' docs like [Nethermind](https://docs.nethermind.io/) or Juno.
- User Guides and Tutorials: Step-by-step guides to help new users get started quickly.
- API References: Detailed and updated documentation of the APIs so developers can easily integrate them.
- FAQ: A helpful section to address common issues and frequently asked questions (I can use the "papyrus-support-only" channel on the Starknet discord as a good source for this)

This would be an open-source contribution, so I do not require any compensation. All I’d need to do is connect with team members to gain more insights into the tool, understand common issues, and gather the necessary information to draft the main sections of the documentation.

I’m excited about the possibility of working on this and contributing to the Starkware community. If you’re interested, I’d love to chat more about how we can get started.

**Reply by iliav-starkware** [reply]
Howdy Oluwabusayo! I'm Ilia from the Starknet product team. I'd love to speak about collaborating. Shooting you an email.

**Reply by TropicolX** [reply]
Hi @iliav-starkware! Picking up from last week, I've sent you a message regarding the docs. Looking forward to hearing from you!

**Comment by daemon2k3** [comment]
Any plans to have a release compatible with the pathfinder 0.14.x version for the upcoming fork ?

**Comment by swift2234** [comment]
Hello, 

I understood correctly that in order to receive rewards, you need to change the wallet addresses in the file?
https://github.com/starkware-libs/papyrus/blob/main/config/default_config.json

"rpc.execution_config.eth_fee_contract_address": {
  "description": "The eth fee token address to receive fees",
  "privacy": "Public",
  "value": "**my_addres_ETH**"
}
......
"rpc.execution_config.strk_fee_contract_address": {
  "description": "The strk fee token address to receive fees",
  "privacy": "Public",
  "value": "**my_addres_STRK**"
}

Thank you for your answer.

**Reply by swift2234** [reply]
I made like 

bro@bro-VirtualBox:~/papyrus/config$ cat my_config.json 
{
  "base_layer.node_url": "http://localhost:8545",
  "base_layer.starknet_contract_address": "0xc662c410C0ECf747543f5bA90660f6ABeBD9C8c4",
  "chain_id": "SN_MAIN",
  "rpc.server_address": "0.0.0.0:8086",
  "rpc.execution_config.strk_fee_contract_address": "0x02606Bc66227f3be0B62dCAa6D9B60631499331c40002Ea966B34bF8baAEa516",
  "rpc.execution_config.eth_fee_contract_address": "0x162F1ADc9bbD648BA4b92B3Aeea43497C464bed9"
}


it is correct ?
